src/spec/base.ts

```diff
@@ -51,11 +51,26 @@ export interface DataMixins {
   data: Data;
 }
 
+export type StepFor = 'position' | 'offset';
+
 export interface Step {
   /**
    * The size (width/height) per discrete step.
    */
   step: number;
+
+  /**
+   * Whether to apply the step to position scale or offset scale when there are both `x` and `xOffset` or both `y` and `yOffset` encodings.
+   */
+  for?: StepFor;
```

Comment:
Any opinion about this property name?

Reply:
`forScale` or `applyTo` or `along` maybe?

Reply:
I think

`{step: 20, for: "position"}`

reads better than 

`{step: 20, forScale/applyTo/along: "position"}`?

Reply:
Full examples

```js
{
  width: {step: 23, for: 'position'},
  mark: 'bar',
  encoding: {
    x: {field: 'x', type: 'nominal'},
    xOffset: {field: 'subx', type: 'nominal'}
  }
}
```

```js
{
  width: {step: 23, for: 'offset'},
  mark: 'bar',
  encoding: {
    x: {field: 'x', type: 'nominal'},
    xOffset: {field: 'subx', type: 'nominal'}
  }
}
```

Another option may be `forChannel`. I am okay with `for`.

Reply:
I think `for` is also better than `forChannel` because it's for position (which is a category of channel).  "for channel position" is swapping the order of the words anyway.  

If it's unclear if a longer word is clearer, I generally prefer brevity